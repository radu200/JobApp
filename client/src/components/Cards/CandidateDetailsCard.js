import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardActionArea from "@material-ui/core/CardActionArea";
import CardContent from "@material-ui/core/CardContent";
import Typography from "@material-ui/core/Typography";
import { Months, Days, Years } from "./../../Utils/messages";
import RoomIcon from "@material-ui/icons/Room";
import Button from '@material-ui/core/Button';

const styles = {
  media: {
    // ⚠️ object-fit is not supported by IE 11.
    objectFit: "cover"
  },

  heading: {
    opacity: 0.7
  },
  candidateAbout: {
    wordWrap: "break-word"
  },
  companyName: {
    color: "black",
    fontWeight: "bold"
  },
  responsibilities: {
    wordWrap: "break-word"
  },
  card: {
    marginTop: 10
  },

  avatar: {
    marginTop: 10,
    marginLeft: 20,
    marginRight: "auto",
    marginBottom: 0,
    width: 200,
    height: 157,
    borderRadius: 10
  },
  lang: {
    fontWeight: "bold"
  },
  availability: {
    color: "#43a047"
  },
  RoomIcon: {
    fontSize: 17
  }
};

const CandidateDetailsCard = ({ candidate, experience, classes}) => {
  return (
    <Card className={classes.card}>
      {candidate.map((candidate, index) => {
        return (
          <CardActionArea key={candidate.id}>
            <img
              alt="Remy Sharp"
              src={candidate.avatar}
              className={classes.avatar}
              />
            <CardContent>
                <Button href={`/api/report/${candidate.id}`} size="small" color="primary">
                  Raporteaza  acest utilizator
                </Button>
              <Typography
                gutterBottom
                component="p"
                className={classes.availability}
              >
                {candidate.job_seeker_availability}
              </Typography>
              <Typography component="p">
                {candidate.job_seeker_employment_type}
              </Typography>
              <Typography gutterBottom variant="h5" component="h2">
                {candidate.first_name} {candidate.last_name}
              </Typography>

              {candidate.job_seeker_about_me ? (
                <Typography component="p" className={classes.candidateAbout}>
                  <b>Despre: </b> {candidate.job_seeker_about_me}
                </Typography>
              ) : null}
              {candidate.job_seeker_languages ? (
                <Typography gutterBottom component="p">
                  <b> Limbi: </b>
                  {candidate.job_seeker_languages}
                </Typography>
              ) : null}

              <Typography component="p">
                <RoomIcon className={classes.RoomIcon} />{" "}
                {candidate.job_seeker_location}
              </Typography>
            </CardContent>
          </CardActionArea>
        );
      })}

      {experience.map((experience, index) => {
        return (
          <CardActionArea key={index}>
            <CardContent>
              <Typography gutterBottom variant="h5" component="h2">
                {experience.position}
              </Typography>
              <Typography component="p">{experience.category}</Typography>
              <Typography component="p" className={classes.companyName}>
                {experience.company_name}
              </Typography>
              <Typography gutterBottom component="p" color="primary">
                {experience.start_date}- {experience.end_date} -{" "}
                {experience.years} {Years} {experience.months} {Months}{" "}
                {experience.days} {Days}
              </Typography>
              <Typography
                gutterBottom
                component="p"
                className={classes.responsibilities}
              >
                {experience.responsibilities}
              </Typography>

            </CardContent>
          </CardActionArea>
        );
      })}
      
         <CardContent>
        </CardContent>
    </Card>
  );
};

CandidateDetailsCard.propTypes = {
  classes: PropTypes.object,
  candidate: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      avatar: PropTypes.string,
      job_seeker_availability: PropTypes.string,
      job_seeker_employment_type: PropTypes.string,
      first_name: PropTypes.string,
      last_name: PropTypes.string,
      job_seeker_about_me: PropTypes.string,
      job_seeker_languages: PropTypes.string,
      job_seeker_location: PropTypes.string.isRequired
    })
  )
};

export default withStyles(styles)(CandidateDetailsCard);
