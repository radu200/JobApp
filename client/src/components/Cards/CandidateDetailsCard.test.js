import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
configure({adapter: new Adapter()});
import CandidateDetailsCard from './CandidateDetailsCard'

describe('CandidateCard',() => {
    it('should render correctly with no props', () => {
        const component = shallow(<CandidateDetailsCard />)
        expect(component).toMatchSnapshot();

    })
    it('should render banner text correctly with given strings', () => {
        const strings = ['one', 'two'];
        const component = shallow(<CandidateDetailsCard list={strings} />);
        expect(component).toMatchSnapshot();
      });
})