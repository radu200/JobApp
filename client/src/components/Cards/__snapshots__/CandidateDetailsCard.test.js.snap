// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CandidateCard should render banner text correctly with given strings 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForwardRef(WithStyles)
    list={
      Array [
        "one",
        "two",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "RoomIcon": "CandidateDetailsCard-RoomIcon-10",
        "availability": "CandidateDetailsCard-availability-9",
        "avatar": "CandidateDetailsCard-avatar-7",
        "candidateAbout": "CandidateDetailsCard-candidateAbout-3",
        "card": "CandidateDetailsCard-card-6",
        "companyName": "CandidateDetailsCard-companyName-4",
        "heading": "CandidateDetailsCard-heading-2",
        "lang": "CandidateDetailsCard-lang-8",
        "media": "CandidateDetailsCard-media-1",
        "responsibilities": "CandidateDetailsCard-responsibilities-5",
      },
      "list": Array [
        "one",
        "two",
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "RoomIcon": "CandidateDetailsCard-RoomIcon-10",
          "availability": "CandidateDetailsCard-availability-9",
          "avatar": "CandidateDetailsCard-avatar-7",
          "candidateAbout": "CandidateDetailsCard-candidateAbout-3",
          "card": "CandidateDetailsCard-card-6",
          "companyName": "CandidateDetailsCard-companyName-4",
          "heading": "CandidateDetailsCard-heading-2",
          "lang": "CandidateDetailsCard-lang-8",
          "media": "CandidateDetailsCard-media-1",
          "responsibilities": "CandidateDetailsCard-responsibilities-5",
        },
        "list": Array [
          "one",
          "two",
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`CandidateCard should render correctly with no props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForwardRef(WithStyles) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "RoomIcon": "CandidateDetailsCard-RoomIcon-10",
        "availability": "CandidateDetailsCard-availability-9",
        "avatar": "CandidateDetailsCard-avatar-7",
        "candidateAbout": "CandidateDetailsCard-candidateAbout-3",
        "card": "CandidateDetailsCard-card-6",
        "companyName": "CandidateDetailsCard-companyName-4",
        "heading": "CandidateDetailsCard-heading-2",
        "lang": "CandidateDetailsCard-lang-8",
        "media": "CandidateDetailsCard-media-1",
        "responsibilities": "CandidateDetailsCard-responsibilities-5",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "RoomIcon": "CandidateDetailsCard-RoomIcon-10",
          "availability": "CandidateDetailsCard-availability-9",
          "avatar": "CandidateDetailsCard-avatar-7",
          "candidateAbout": "CandidateDetailsCard-candidateAbout-3",
          "card": "CandidateDetailsCard-card-6",
          "companyName": "CandidateDetailsCard-companyName-4",
          "heading": "CandidateDetailsCard-heading-2",
          "lang": "CandidateDetailsCard-lang-8",
          "media": "CandidateDetailsCard-media-1",
          "responsibilities": "CandidateDetailsCard-responsibilities-5",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
